File: deposito/AbstractHTMLImageProducer.java
package giada.swing.MnR;

import simulation.dom.$Image;

/**
 * The abstract object able to produce an HTML image element to use in an
 * AbstractSliderModelAndRenderer
 *
 * @author gianpiero.diblasi
 * @param <T> The value type of the AbstractSliderModelAndRenderer
 */
public abstract class AbstractHTMLImageProducer<T> {

  private final T value;

  public AbstractHTMLImageProducer(T value) {
    this.value = value;
  }

  public abstract $Image produce();

  public T getValue() {
    return this.value;
  }
}

File: transpile/javascript/util/AbstractHTMLImageProducer.js
/**
 * The abstract object able to produce an HTML image element based on a value
 *
 * @author gianpiero.diblasi
 * @param <T> The value type
 */
class AbstractHTMLImageProducer {

   value = null;

  /**
   * Creates the object
   *
   * @param value The value
   */
  constructor(value) {
    this.value = value;
  }

  /**
   * Produces an HTML image element
   *
   * @return An HTML image element
   */
   produce() {
  }

  /**
   * Returns the value
   *
   * @return The value
   */
   getValue() {
    return this.value;
  }
}
